src/common/FileSystem.cpp

```diff
@@ -2332,25 +2332,28 @@ std::string DefaultHomePath()
 {
 #ifdef _WIN32
 	wchar_t buffer[MAX_PATH];
-	if (!SUCCEEDED(SHGetFolderPathW(nullptr, CSIDL_PERSONAL, nullptr, SHGFP_TYPE_CURRENT, buffer)))
-		return "";
-	return Str::UTF16To8(buffer) + "\\My Games\\" PRODUCT_NAME;
-#else
+	if (SUCCEEDED(SHGetFolderPathW(nullptr, CSIDL_PERSONAL, nullptr, SHGFP_TYPE_CURRENT, buffer))) {
+		return Path::Build(Path::Build(Str::UTF16To8(buffer), "My Games"), PRODUCT_NAME);
+	}
+#elif defined(__APPLE__)
 	const char* home = getenv("HOME");
-	if (!home)
-		return "";
-#ifdef __APPLE__
-	return std::string(home) + "/Library/Application Support/" PRODUCT_NAME;
+	if (home && !!home[0]) {
+		return Path::Build(Path::Build(Path::Build(home, "Library"), "Application Support"), PRODUCT_NAME);
+	}
 #else
-	const char* _xdgDataHome = getenv("XDG_DATA_HOME");
-	std::string xdgDataHome = _xdgDataHome == nullptr ? Path::Build(Path::Build(std::string(home), ".local") ,"share") : std::string(_xdgDataHome);
-	std::string xdgHomePath;
-
-	xdgHomePath = Path::Build(xdgDataHome, PRODUCT_NAME_LOWER);
-
-	return xdgHomePath;
-#endif
+	const char* xdgDataHome_ = getenv("XDG_DATA_HOME");
+	if (xdgDataHome_ && !!xdgDataHome_[0]) {
+		return Path::Build(xdgDataHome_, PRODUCT_NAME_LOWER);
+	}
+	else {
+		const char* home = getenv("HOME");
+		if (home && !!home[0]) {
```

Comment:
Why do you need the !!. `if (home && home[0])` will work just fine?

Reply:
Well, it's pedantic, `home[0]` is a `char` while `!!home[0]` is a `bool`, but I can remove that.

Reply:
Done.